src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1318,61 +1318,93 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
+            bool? resultIsNullable;
             if (IsConstantTrue(node.Condition))
             {
-                VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
-                VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
-                // it may be a boolean state at this point.
+                SetState(consequenceState);
+                resultIsNullable = getIsNullable(consequenceResult);
             }
             else if (IsConstantFalse(node.Condition))
             {
-                VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
-                VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
-                // it may be a boolean state at this point.
+                resultIsNullable = getIsNullable(alternativeResult);
             }
             else
             {
-                VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
-                Unsplit();
-                var consequenceType = _result.Type;
-                consequenceState = this.State;
-                VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
-                Unsplit();
-                var alternativeType = _result.Type;
                 IntersectWith(ref this.State, ref consequenceState);
-                if ((object)consequenceType == null || (object)alternativeType == null || !consequenceType.Equals(alternativeType, TypeCompareKind.ConsiderEverything))
+                resultIsNullable = (getIsNullable(consequenceResult) | getIsNullable(alternativeResult));
+            }
+
+            TypeSymbolWithAnnotations resultType;
+            if (consequence.Kind == BoundKind.ThrowExpression)
+            {
+                resultType = alternativeResult.Type;
+            }
+            else if (alternative.Kind == BoundKind.ThrowExpression)
+            {
+                resultType = consequenceResult.Type;
+            }
+            else
+            {
+                if (node.HasErrors)
                 {
-                    var consequenceIsNullable = (object)consequenceType == null ? true : consequenceType.IsNullable;
-                    var alternativeIsNullable = (object)alternativeType == null ? true : alternativeType.IsNullable;
-                    _result = TypeSymbolWithAnnotations.Create(node.Type, isNullableIfReferenceType: consequenceIsNullable | alternativeIsNullable);
+                    resultType = null;
                 }
                 else
                 {
-                    // PROTOTYPE(NullableReferenceTypes): Use BestTypeInferrer.InferBestTypeForConditionalOperator
-                    // to ensure nested nullability is considered.
-                    _result = GetMoreNullableType(consequenceType, alternativeType);
+                    BoundExpression createPlaceholderIfNecessary(BoundExpression expr, Result result)
```

Comment:
Consider moving local function at the end of the method, for readability. Thanks